app/services/api/public/node/filter.rb

```diff
@@ -0,0 +1,125 @@
+module Api
+  module Public
+    module Node
+      class Filter
+        attr_reader :data
+
+        def initialize(params = {})
+          initialize_params(params)
+        end
+
+        def call
+          initialize_query
+
+          apply_filters
+
+          @query.first
+        end
+
+        private
+
+        def initialize_params(params)
+          @node_id = params[:id]
+          @year = params[:year]
+        end
+
+        def initialize_query
+          @query = Api::V3::Readonly::FlowNode.
+            select(*select_clause).
+            joins('INNER JOIN nodes_mv ON nodes_mv.id = flow_nodes_mv.node_id').
+            joins('INNER JOIN contexts_mv ON contexts_mv.id = flow_nodes_mv.context_id').
+            group(*group_clause).
+            order('flow_nodes_mv.node_id')
+        end
+
+        def select_clause
+          [
+            'flow_nodes_mv.node_id',
+            'nodes_mv.name',
+            'nodes_mv.node_type',
+            'nodes_mv.geo_id',
+            'JSON_AGG(' \
+              'DISTINCT JSONB_BUILD_OBJECT(' \
+                '\'country\', contexts_mv.iso2, ' \
+                '\'commodity\', contexts_mv.commodity_name, ' \
+                '\'years\', nodes_mv.years' \
+              ')' \
+            ') AS availability',
+            'JSON_AGG(' \
+              'DISTINCT JSONB_BUILD_OBJECT(' \
+                '\'country\', contexts_mv.iso2, ' \
+                '\'commodity\', contexts_mv.commodity_name, ' \
+                "'node_quants', #{select_node_clause('quants')}, " \
+                "'node_quals', #{select_node_clause('quals')}, " \
+                "'node_inds', #{select_node_clause('inds')}" \
+              ')' \
+            ') AS node_indicators',
+            'JSON_AGG(' \
+              'DISTINCT JSONB_BUILD_OBJECT(' \
+                '\'country\', contexts_mv.iso2, ' \
+                '\'commodity\', contexts_mv.commodity_name, ' \
+                '\'flow_id\', flow_nodes_mv.flow_id, ' \
```

Comment:
we probably don't need the flow_id? I think it's not necessary to include that in the response, let's not expose any internal identifiers unless strictly necessary. for example, we expose the node ids in the nodes endpoints, because that allows the user to use those ids as filters in other endpoints. I think we don't currently have anywhere where they could filter by flow, and probably we won't have.